packages/aws-cdk-lib/aws-events-targets/lib/ecs-task.ts

```diff
@@ -277,10 +277,18 @@ export class EcsTask implements events.IRuleTarget {
   }
 
   private createEventRolePolicyStatements(): iam.PolicyStatement[] {
+    // check if there is a taskdefinition revision (arn will end with : followed by digits) included in the arn already
+    let needsRevisionWildcard = false;
+    if ( !cdk.Token.isUnresolved(this.taskDefinition.taskDefinitionArn)) {
+      const revisionAtEndPattern = /:[0-9]+$/;
+      const hasRevision = revisionAtEndPattern.test(this.taskDefinition.taskDefinitionArn);
+      needsRevisionWildcard = !hasRevision;
+    }
```

Comment:
This code isn't needed. As mentioned in https://github.com/aws/aws-cdk/issues/30751, the task definition without the trailing `:*` isn't a valid ARN. Verified in the Console:

<img width="686" alt="Screenshot 2024-10-03 at 6 20 50 PM" src="https://github.com/user-attachments/assets/9cc7e329-3994-488f-9fea-d7431b38c479">

Reply:
Hi @msambol , thanks for taking time to review. It is possible for `taskDefinitionArn` to have a revision number that isn't `*`, hence, this `if` block. When there is already a revision number, we shouldn't append `:*`. I left a comment in the [PR](https://github.com/aws/aws-cdk/pull/30788) you linked about this case.

Reply:
The `*` means it will match revision numbers (it's a wildcard).

Reply:
Yes. That is correct. It is also possible that customer may want to restrict the IAM policy resource to specific revision number.

Reply:
Obviously different purpose of the function but similar pattern.

Reply:
Moving it out to a public method to allow unit testing means exposing it for potential public use, which will become a API contract that cannot have any breaking change. I would prefer keeping it private and local in this method for now. The unit tests added in this PR are primarily focusing on this block of code.